### PR TITLE
Add active date ranges to lms courses

### PR DESCRIPTION
**Pull request by dtgreiner**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch
- use a merge-commit or rebase strategy for PRs targeting the stable branch

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)

Add date ranges where TalentLMS courses will be considered active. Only active courses will be included in the required training for individual users.

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)

New feature

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] If adding a new endpoint / exposing data in a new way, I have:
  - [ ] ensured the API can't leak data from other data sources
  - [ ] ensured this does not introduce N+1s
  - [ ] ensured permissions and visibility checks are performed in the right places
- [ ] I have updated the documentation (or not applicable)
- [ ] I have added spec tests (or not applicable)
- [X] I have provided testing instructions in this PR or the related issue (or not applicable)

[//]: # NOTE: system tests may fail if there is no branch on the hmis-frontend that matches the Source  or Target branch of this PR. This is expected

### CODE REVIEW

File: app/models/talentlms/course.rb [hunk 1]
```diff
@@ -68,5 +85,14 @@ def completed_by?(user)
 
       Talentlms::CompletedTraining.where(login_id: login.id, course_id: id).exists?
     end
+
+    def active_date_range_for_display
+      return start_date if start_date.present? && end_date.present? && start_date == end_date
```
Comment (eanders) [inline app/models/talentlms/course.rb:90]: This should be impossible based on the validation that ensures `end_date > start date` (also if needed I think you can just use: `start_date.present? && start_date == end_date`

File: app/controllers/admin/talentlms_courses_controller.rb [hunk 2]
```diff
@@ -46,6 +46,8 @@ def course_params
         :courseid,
         :months_to_expiration,
         :default,
+        :start_date,
```
Comment (eanders) [inline app/controllers/admin/talentlms_courses_controller.rb:49]: I can't comment above, but the courses should probably be sorted in some way that places the active courses first in the table in date order.

**Review by eanders — CHANGES_REQUESTED** [review]
Thanks @dtgreiner.  This looks great, and thanks for adding the validations!
Couple comments inline that I think will make this a bit better.

File: app/controllers/admin/talentlms_courses_controller.rb (hunk 2 above)
Comment (dtgreiner) [reply app/controllers/admin/talentlms_courses_controller.rb:49]: I'll plan to add some sorting to the course to address this

File: app/models/talentlms/course.rb (hunk 1 above)
Comment (dtgreiner) [reply app/models/talentlms/course.rb:90]: This should be able to occur. The validation should only be firing if both start and end dates are present and start date is > end date. It should be valid to have it start and end on the same date, but I'll double check the logic to be sure.

Comment (eanders) [reply app/models/talentlms/course.rb:90]: Ah, that makes sense.  Disregard.

**Comment by dtgreiner** [comment]
@eanders I have a different PR for this [here](https://github.com/greenriver/hmis-warehouse/pull/4973). This one had a bad merge when trying to get the latest into the branch and it was easier to recreate.

**Pull request closed without merging**